CMakeLists.txt

```diff
@@ -9,38 +9,42 @@ set(CMAKE_MODULE_PATH
 # User-settings
 option(SERIOUSPROTON_WITH_JSON "Use json library." OFF)
 
+#
+set(EXTERNALS_DIR "${PROJECT_BINARY_DIR}/externals")
+set(DOWNLOADS_DIR "${PROJECT_BINARY_DIR}/downloads")
+file(MAKE_DIRECTORY "${EXTERNAL_DIR}" "${DOWNLOADS_DIR}")
```

Comment:
Inserting this here kinda breaks my initial intent of grouping options near the top, to make them easier to locate.

Not that it's a huge deal though.

Reply:
I am used to find options / variables using  ```cmake -LAH```
If it is not present there, then I look into the code. Either the variable is there but is missing CACHE attribute; or such option / variable is missing entirely.

Asking CMake cache to show variables also makes some difference between "local and global" variables. (I mean, some variables are only meant to be used locally in the code and should never be configured by the user)

I see a value in keeping a code regarding one functionality together.